src/qt/walletcontroller.cpp

```diff
@@ -262,9 +262,12 @@ void CreateWalletActivity::createWallet()
     }
 
     QTimer::singleShot(500ms, worker(), [this, name, flags] {
-        std::unique_ptr<interfaces::Wallet> wallet = node().walletLoader().createWallet(name, m_passphrase, flags, m_error_message, m_warning_message);
+        auto wallet{node().walletLoader().createWallet(name, m_passphrase, flags, m_warning_message)};
 
-        if (wallet) m_wallet_model = m_wallet_controller->getOrCreateWallet(std::move(wallet));
+        m_error_message = util::ErrorString(wallet);
+        if (wallet) {
+            m_wallet_model = m_wallet_controller->getOrCreateWallet(std::move(*wallet));
+        }
```

Comment:
why the revert here? (and in the others as well).
the result could be an object or an error, not both.

```c++
if (wallet) m_wallet_model = something;
else m_error_message = util::ErrorString(wallet);
```

Thinking that might be good to add an assertion inside `util::ErrorString`, callers should always check that the result is an error prior to convert it.

Reply:
Maybe `assert(std::holds_alternative<bilingual_str>(result.m_variant));` inside `util::ErrorString` ?

However this assertion cause a compile-time error `error: static_assert failed due to requirement '__detail::__variant::__exactly_once<bilingual_str, bilingual_str, bilingual_str>' "T must occur exactly once in alternatives"`

Reply:
yep, that was where I was pointing to.
`assert(!result.has_value());` there should work fine.

Reply:
I applied the suggestion above (`if (wallet) m_wallet_model = something;`) in https://github.com/bitcoin/bitcoin/pull/25616/commits/ac7479953bc95baf6ef4b35fa25c27dd4100fe6a.

But regarding the assertion, I think a follow-up PR is better  because it also requires changing the `src/test/result_tests.cpp` which expects the result of `util::ErrorString` be `{}` if the object has value This would increase the scope of this PR.

Reply:
Missing brackets in the conditional again (e.g. https://github.com/bitcoin/bitcoin/pull/25616#discussion_r938595550)

Reply:
shouldn't be needed:
https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md?plain=1#L77-L80

Reply:
"If an if only has a single-statement then-clause, it can appear on the same line as the if, without braces. In every other case, braces are required, and the then and else clauses must appear correctly indented on a new line."

Reply:
yep, aren't we having a single line statement here?

```c++
if (wallet) m_wallet_model = m_wallet_controller->getOrCreateWallet(std::move(*wallet));
else m_error_message = util::ErrorString(wallet);
```